types/types.go

```diff
@@ -235,19 +247,16 @@ func (typ Type) Label() string {
 		return "EMPTY"
 	}
 
-	if typ[0]&'\xf0' == '\x00' {
```

Comment:
note: I changed the handling here because it was forcing the "simple types" to remain in the last few remaining slots (low numbers). It didn't really give us a big advantage, so instead we are now looking it up as a simple type first and as a complex one second.  `labelfun` is still needed for the more complex types that need a bit of handling for their label